File: jung-io/src/test/java/edu/uci/ics/jung/io/graphml/DummyEdge.java
package edu.uci.ics.jung.io.graphml;

import java.util.function.Function;

public class DummyEdge extends DummyGraphObjectBase {

  public static class EdgeFactory implements Function<EdgeMetadata, DummyEdge> {
    int n = 100;

    public DummyEdge apply(EdgeMetadata md) {
      return new DummyEdge(n++);
    }
  }

  public DummyEdge() {}

  public DummyEdge(int v) {
    super(v);
  }
}

File: jung-io/src/test/kotlin/edu/uci/ics/jung/io/graphml/DummyEdge.kt
package edu.uci.ics.jung.io.graphml

import java.util.function.Function

class DummyEdge : DummyGraphObjectBase {

  class EdgeFactory : Function<EdgeMetadata, DummyEdge> {
    var n = 100

    override fun apply(md: EdgeMetadata): DummyEdge {
      return DummyEdge(n++)
    }
  }

  constructor() : super()

  constructor(v: Int) : super(v)
}
